### ISSUE TITLE
Skeleton for new Apm Java agent

### ISSUE THREAD

**Issue by jackshirazi** [elastic/observability-docs#3558]
### Description

We're adding a new APM Java agent based on the OpenTelemetry agent, so it will be a distribution of the OpenTelemetry Java agent with additional Elastic extensions. We'll aim to generate documentation within the repo (elastic-otel-java) which should be deployed to the website similarly to the other Elastic agents - but if there's a better way, we're happy to adopt that. The existing Elastic Java agent documentation is [here](https://www.elastic.co/guide/en/apm/agent/java/current/index.html) and there's the question about how this should be displayed in relation to that. This is a completely new agent, but it's also a Java agent, just a different implementation

### Resources

https://github.com/elastic/observability-dev/issues/2882

### Which documentation set does this change impact?

Integrations

### Feature differences

n/a

### What release is this request related to?

N/A

### Collaboration model

The engineering team

### Point of contact.

**Main contact:** @jackshirazi 

**Stakeholders:** @elastic/apm-agent-java

**Comment by bmorelli25** [comment]
Hi Jack. My team would be happy to help with this. A few dumb questions to kick things off:

**Where should the new docs live?**
* _Option 1_: Add documentation for the new agent to the existing [APM Java Agent](https://www.elastic.co/guide/en/apm/agent/java/current/index.html) book.
* _Option 2_: Create a new documentation book exclusively for the new APM Java agent.

Based on what I know about this project, I think the new Otel agent is significantly different from the existing agent, which makes option 2 the only viable option. Does that sound right to you? If we establish this pattern for the Java Otel Elastic agent, is it a pattern we can follow for the other language otel agents?

**Where should the new "book" live?**

Our current document system has the concept of "books". In Observability, we have 13 books—one for Observability, ten for APM agents, and two for extensions. You can see an overview of all Observability books [on this page](https://www.elastic.co/guide/en/apm/index.html).

I'm assuming that because these are different products, we want to distinguish between pure Elastic APM agents and Elastic OTel agents. Is that correct? Ignoring the likely incorrect naming, something like this?

<img width="376" alt="Screenshot 2024-03-26 at 1 31 17 PM" src="https://github.com/elastic/observability-docs/assets/5618806/b508b099-e1e4-4a44-b9da-87362371f0c7">

The other way to go about this would be to add the OTel distributions to the same list:

<img width="340" alt="Screenshot 2024-03-26 at 1 32 52 PM" src="https://github.com/elastic/observability-docs/assets/5618806/b11bb248-f30b-4a33-9b82-a8ca0c25a9c0">

If this makes more sense to you, I'd be curious to understand why.

**Looking forward**

The book-based architecture of our current documentation system is very limiting and leads to a segmented documentation experience. In late April/early May, we're going to migrate Observability documentation to our new system. This system completely does away with book-based architecture and allows us to better connect and display different documentation sets. For example, here's a look at what the high-level Observability documentation experience will be:

![Screenshot 2024-03-26 at 10 02 58 AM](https://github.com/elastic/observability-docs/assets/5618806/1d0d6b01-8821-4088-a3ae-4bfa9b2f9f58)

On a single page, a user will be able to navigate between APM agent documentation sets and all other relevant Observability documentation, like the [Observability Guide](https://www.elastic.co/guide/en/observability/current/apm.html), without losing navigation context.

Since we're just a month or two away from living in this new documentation world, I'd like to keep the information architecture of the new system in mind as we make decisions about where this content will live in our current documentation system.

I have a couple of additional questions that I'll ask in a private channel.

**Comment by jackshirazi** [comment]
> Based on what I know about this project, I think the new Otel agent is significantly different from the existing agent, which makes option 2 the only viable option. Does that sound right to you? If we establish this pattern for the Java Otel Elastic agent, is it a pattern we can follow for the other language otel agents?

Yes, I agree it's a significantly different agent. All of the options will be different (and we'll need to explain to users how to adapt old options to new options and the incompatibilities they have)

> something like this?
> 
> <img alt="Screenshot 2024-03-26 at 1 31 17 PM" width="376" src="https://private-user-images.githubusercontent.com/5618806/317030177-b508b099-e1e4-4a44-b9da-87362371f0c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE0OTAzMjAsIm5iZiI6MTcxMTQ5MDAyMCwicGF0aCI6Ii81NjE4ODA2LzMxNzAzMDE3Ny1iNTA4YjA5OS1lMWU0LTRhNDQtYjlkYS04NzM2MjM3MWYwYzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjZUMjE1MzQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGVlZTFmYzRjM2RkN2U1YzYxZjBjZDgzYzVlMmViZGE4MTViOTE3YTU2YjFmYmZiODBhMGZmZmM0MWMzMjBkNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.-3hlUR2UNuBjxvc9fe7FGVQIgJWCp1EPB6NLU7Nr9l8">

I agree. Maybe the other way round, the future is the otel distros

**Comment by bmorelli25** [comment]
I was talking with Trent this afternoon and he brought up an intriguing third option: organize the nav by language. I think this option makes a lot of sense. Here's what it might look like after our migration to Docsmobile:

<img width="1312" alt="potential-nav-structure-highlight" src="https://github.com/elastic/observability-docs/assets/5618806/8a9592ee-ef55-44b6-ac85-9ac795cd37ac">

**Comment by stevejgordon** [comment]
> I was talking with Trent this afternoon and he brought up an intriguing third option: organize the nav by language.

I like this idea as it's quicker for a language developer to find all relevant content. It also aligns more closely with how the OTel docs are organized.

**Comment by trentm** [comment]
> It also aligns more closely with how the OTel docs are organized.

Good guess on where I got the idea. :)

**Comment by trentm** [comment]
@bmorelli25 If we can pull that off for the new docs, then that would be great.

For the current docs, I think either would be fine (especially given these docs may be short-lived), but I would bias to your first suggestion of separating the "Elastic APM agents" block from the "Elastic OpenTelemetry language distributions" block.

**Comment by bmorelli25** [comment]
> For the current docs, I think either would be fine (especially given these docs may be short-lived), but I would bias to your first suggestion of separating the "Elastic APM agents" block from the "Elastic OpenTelemetry language distributions" block.

That's what I'm thinking, too. Since we're so close to migration, there's no point in rearranging the current docs. I've [opened a PR](https://github.com/elastic/docs/pull/2972) that will add a new section above the existing APM Agent section for `OpenTelemetry Distributions`.  Then, after we've migrated to the new documentation system, we'll break apart the content as demonstrated in https://github.com/elastic/observability-docs/issues/3558#issuecomment-2021803984.

That brings up the next topic: naming. Have we already identified a pattern for naming these new agents/distributions/libraries/tools? @trentm pointed out the following:

> https://opentelemetry.io/docs/languages/ calls these variously "OpenTelemetry code instrumentation support for [language]", "A language-specific implementation of OpenTelemetry in [language]", etc. For some languages (e.g. Java, sometimes Python) the term "agent" meaningfully applies to one part of this language support.

**Comment by trentm** [comment]
> That brings up the next topic: naming. Have we already identified a pattern for naming these new agents/distributions/libraries/tools?

Yes. At least us dev-ies discussed this on our weekly group call this week. 

```
Required elements for disambiguation:
1. Elastic
2. OpenTelemetry (or OTel for brevity/less formality)
3. (Language)
4. Distribution (or “distro” for brevity or with less formality)
```

To disambiguate one of these things requires all four of those elements. For example the "Elastic OpenTelemetry Node.js Distribution".  Current doc context might mean some could be elided. We didn't specify whether the order is strongly required. E.g. "Elastic .NET OpenTelemetry Distro" is just fine, IMHO.

The "(Language)" component is to disambiguate both (a) between the language distros and (b) distros of other OTel components, like the OTel Collector.

I am morbidly curious if:

```
E-las-tic O-pen-te-le-me-try Node-dot-j-s Dis-tri-bu-tion
1   2   3 4   5  6  7  8   9   10  11 . .  14  15 16   17
```

17 syllables is a record for a "product" name.

**Comment by bmorelli25** [comment]
That is a mouthful, but I see why all four components are included in the name. Thanks. It also looks like other folks came to the same conclusion (although they also include `for`):

* [Honeycomb OpenTelemetry Distro for Node.js](https://github.com/honeycombio/honeycomb-opentelemetry-node)
* [Splunk Distribution of OpenTelemetry for Node.js](https://github.com/signalfx/splunk-otel-python)
* [Lumigo OpenTelemetry Distro for Node.js](https://github.com/lumigo-io/opentelemetry-js-distro)

I must admit, having the language name at the end does IMO help it stick out more:
```
Elastic Java OpenTelemetry Distribution
Elastic OpenTelemetry Distribution for Node.js
```

**Comment by trentm** [comment]
> I must admit, having the language name at the end does IMO help it stick out more:

Works for me.

**Comment by stevejgordon** [comment]
@bmorelli25 What are the next steps to move this forward?

**Comment by bmorelli25** [comment]
Are there docs that are ready to be published? If so, I can set up the infra side of things and add the otel repo to our doc build.

**Comment by stevejgordon** [comment]
@bmorelli25 We have [initial getting started docs](https://github.com/elastic/elastic-otel-dotnet/tree/main/docs) for .NET that's ready. It relates to our alpha release, so it lives in the main at the moment. If we need to cut a version branch, we can do so.

**Comment by colleenmcginnis** [comment]
All OTel repos have docs. 🎉 We also aligned on naming and terminology, and I opened a PR to add guidelines to our internal docs.
